packages/react/src/components/Combobox/Combobox.tsx

```diff
@@ -235,16 +335,33 @@ const Combobox = forwardRef<HTMLDivElement, ComboboxProps>(
         onBlur?.(event);
         setOpen(false);
         if (autocomplete === 'automatic' && activeDescendant) {
-          const stringValue =
+          const activeValue =
             activeDescendant.value?.toString() ||
             /* istanbul ignore next: default value */ '';
-          setValue(stringValue);
-          setSelectedValue(stringValue);
-          onSelectionChange?.({
-            target: activeDescendant.element,
-            value: stringValue,
-            previousValue: value
-          });
+          setInputValue(activeValue);
+
+          if (!multiselect) {
+            setSelectedValues([activeValue]);
+            (onSelectionChange as OnSingleSelectionChange)?.({
+              target: activeDescendant.element,
+              value: activeValue,
+              previousValue: selectedValues[0]
+            });
+          } else {
+            const idx = selectedValues.indexOf(activeValue);
+            const newSelectedValues = selectedValues.filter(
+              (v) => v !== activeValue
+            );
+            if (idx === -1) {
+              newSelectedValues.push(activeValue);
+            }
+            setSelectedValues(newSelectedValues);
+            (onSelectionChange as OnMultiSelectionChange)?.({
+              target: activeDescendant.element,
+              value: newSelectedValues,
+              previousValue: selectedValues
+            });
```

Comment:
Nit: We should avoid abbreviated names when possible. Also providing a descriptive value helps for readability since we know what index it is for.

```suggestion
            const activeValueIndex = selectedValues.indexOf(activeValue);
            const newSelectedValues = selectedValues.filter(
              (v) => v !== activeValue
            );
            if (activeValueIndex === -1) {
              newSelectedValues.push(activeValue);
            }
            setSelectedValues(newSelectedValues);
            (onSelectionChange as OnMultiSelectionChange)?.({
              target: activeDescendant.element,
              value: newSelectedValues,
              previousValue: selectedValues
            });
```

Reply:
Updated